src/main/java/com/mongodb/kafka/connect/util/ConnectionValidator.java

```diff
@@ -0,0 +1,183 @@
+
+            List<String> unsupportedUserActions = new ArrayList<>(actions);
+            for (final Document userInfo : usersInfo.getList("users", Document.class)) {
+                unsupportedUserActions = removeUserActions(userInfo, databaseName, collectionName, actions);
+                if (!unsupportedUserActions.isEmpty() && userInfo.getList("inheritedPrivileges", Document.class).isEmpty()) {
+                    for (final Document inheritedRole : userInfo.getList("inheritedRoles", Document.class)) {
+                        String roleDB = inheritedRole.getString("db");
+                        String roleName = inheritedRole.getString("role");
+                        Document rolesInfo = mongoClient.getDatabase(roleDB)
+                                .runCommand(Document.parse(format("{rolesInfo: '%s', showPrivileges: 1, showBuiltinRoles: 1}", roleName)));
+                        for (final Document roleInfo : rolesInfo.getList("roles", Document.class)) {
+                            unsupportedUserActions = removeUserActions(roleInfo, databaseName, collectionName, unsupportedUserActions);
+                        }
+
+                        if (unsupportedUserActions.isEmpty()) {
+                            return;
+                        }
+                    }
+                }
+                if (unsupportedUserActions.isEmpty()) {
+                    return;
+                }
+            }
+
+            String missingPermissions = String.join(", ", unsupportedUserActions);
+            getConfigByName(config, configName).ifPresent(c ->
+                    c.addErrorMessage(format("Invalid user permissions. Missing the following action permissions: %s", missingPermissions))
+            );
+        } catch (MongoSecurityException e) {
+            getConfigByName(config, configName).ifPresent(c ->
+                    c.addErrorMessage(format("Invalid user permissions authentication failed."))
+            );
+        } catch (Exception e) {
+            LOGGER.warn("Permission validation failed due to: {}", e.getMessage(), e);
+        }
+    }
+
+    private static List<String> removeUserActions(final Document rolesInfo, final String databaseName, final String collectionName,
+                                          final List<String> userActions) {
+        List<Document> privileges = rolesInfo.getList("inheritedPrivileges", Document.class);
+        if (privileges.isEmpty() || userActions.isEmpty()) {
+            return userActions;
+        }
+
+        List<String> unsupportedUserActions = new ArrayList<>(userActions);
+        for (final Document privilege : privileges) {
+            Document resource = privilege.get("resource", Document.class);
+            String database = resource.getString("db");
+            String collection = resource.getString("collection");
+
+            if (database.isEmpty() || (database.equals(databaseName) && collection.isEmpty())
```

Comment:
Here if the db / collection field is empty it means all dbs/collection - so we have to check we match the databaseName and collectionName against empty values as well